src/file_functions.R

```diff
@@ -121,3 +121,45 @@ sf_to_zip <- function(zip_filename, sf_object, layer_name){
   zip(file.path(cdir, zip_filename), files = files_to_zip)
   setwd(cdir)
 }
+
+# bring together various water level sources
+combine_level_sources <- function(out_rds, nwis_levels, nyc_levels, hist_levels) {
+  nwis <- readRDS(nwis_levels)
+  nyc <- readRDS(nyc_levels)
+  hist <- readRDS(hist_levels) %>% 
+    mutate(date = as.Date(date)) %>%
+    rename(surface_elevation_m = res_level_m) %>%
+    # filtering out monthly interpolated values since we now of NYC-DEP data that covers same period
+    filter(!data_type %in% 'monthly')
+  
+  obs_dat <- bind_rows(nwis, nyc, hist, .id = 'source') %>%
+    group_by(site_id, date) %>%
+    # sources were ordered by priority, so can filter on min source
+    slice_min(source) %>%
```

Comment:
Haven't seen `slice_min()` before -- cool how it pairs with the `group_by` call

Reply:
Ooh, yeah, that's news to me too! Nice!